protofile/file_linux.go

```diff
@@ -43,11 +42,11 @@ func (p generalizedProtoFile) SizeWillBe(numBytes uint64) error {
 
 	fd := int(p.File.Fd())
 	if numBytes <= maxInt64 {
-		return syscall.Fallocate(fd, 0, 0, int64(numBytes))
+		return fallocate(fd, 0, int64(numBytes))
 	}
-	err := syscall.Fallocate(fd, 0, 0, maxInt64)
+	err := fallocate(fd, 0, maxInt64)
 	if err != nil {
 		return err
 	}
-	return syscall.Fallocate(fd, 0, maxInt64, int64(numBytes-maxInt64))
+	return fallocate(fd, maxInt64, int64(numBytes-maxInt64))
```

Comment:
If the first `fallocate` failed (silently) there's no reason why we should issue the syscall a second time.

Syscalls are costly: They stop the entire thread, and a context switch happens. They're only worthwile if we get functionality in return.